Improvements

FedG-code opened the pull request:
Better text formatting with margin usage. A header that at the moment doesn't take up the full page. A broken progress bar, I think my script might be in the wrong spot?

FedG-code committed:
Improvements

Better text formatting with margin usage. A header that at the moment doesn't take up the full page. A broken progress bar, I think my script might be in the wrong spot?

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,107 @@
+      <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>{{{title}}}</title>
```
this can be only `{{` triple is for escaping HTML

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,107 @@
+
+<body>
+  <script>
+  $(document).on('ready',function{
```
this is jquery notation you want something like
```
document.addEventListener("DOMContentLoaded", function(event) {
    // Your code to run since DOM is loaded and ready
});
```
However I would recommend loading this script at the end, then all the elements are ready and you dont need this

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,107 @@
+    const myHTML = document.documentElement;
+    const documentHeight = Math.max(documentBody.scrollHeight, documentBody.offsetHeight, myHTML.clientHeight, myHTML.scrollHeight, myHTML.offsetHeight);
+
+    progressBar = $('progress'), max, value;
```
anything with `$()` is jquery but I'm also not really sure what this: 
``
progressBar = $('progress'), max, value;
```

Is trying to do

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,107 @@
+
+    progressBar = $('progress'), max, value;
+    var max =documentHeight -windowHeight; 
+    $("progress").attr('max', max);
```
again jquery, see: 
https://www.javatpoint.com/javascript-getattribute-method#:~:text=%E2%86%92%20%E2%86%90%20prev-,JavaScript%20getAttribute()%20method,an%20empty%20string%20or%20null.

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,107 @@
+    var max =documentHeight -windowHeight; 
+    $("progress").attr('max', max);
+  
+    $(document).on('scroll', function(){
```
jquery

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,107 @@
+    $("progress").attr('max', max);
+  
+    $(document).on('scroll', function(){
+      var value = $(window).scrollTop(); 
```
you get what I'm saying :)

robinglen commented on package.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": {
+    "dot": "^1.1.3",
```
remove this

robinglen commented on blog/converter.js:
```diff
@@ -1,19 +1,37 @@
+    var titleFull = newHTMLFile.substring(titleStart, titleEnd);
+    //remove the header formatting
+    titleFull = titleFull.replace(jsHeaderStart, '');
+    titleFull = titleFull.replace(jsHeaderEnd, '');
```
You could also, might be easier get the title from the filename and then sentence case  and clean up... or even easier use a regex with a capture group to find the `<h1>*</h1>`

FedG-code committed:
Successful progress bar

Scrolling bar works but it's in the wrong place. The banner itself still isn't taking up the whole screen. I can't seem to find a way to isolate the margin to just the text I bring in.

FedG-code committed:
Updated images cleaner look

robinglen commented:
Ok from a design view this is what I would do
![image](https://user-images.githubusercontent.com/8751165/194922433-992af36b-699f-49c6-aba5-4b3d2b747a4e.png)
* Make the header bar blue use the homepage BG blue
* Use the all white gullwing logo center aligned and much smaller should link to the homepage
* On the left side we should add a new icon thats like back to the new blog index we discussed, use placeholder for now and I will get James to think of something
* The progress bar should be 100% width and at the bottom of the sticky header
* It should go from white to the red of the logo
* Image widths need to be fixed to the container they are in, see: 
![image](https://user-images.githubusercontent.com/8751165/194923064-d811d665-249d-45af-a3d4-0df9f3cd3717.png)

* Body copy I think needs to lose bold
* We need to style `a` tags, I think we try the blue from the header
* fix the title
```
<title> id&#x3D;&quot;beyondgitopshowwereleaseourmicroservicesonkubernetesatynap&quot;&gt;Beyond GitOps: How we release our Microservices on Kubernetes at YNAP</title>
```
* Add a description, it could be the first 100 words then `...` shouldn't include the heading title in it
* Open Graph tags should be included
* Generate an index file that should be links to the stories

FedG-code committed:
Fixed header colour and allignment

FedG-code committed:
Header fixed

FedG-code committed:
Added list.html as a storage for the blog links

As long as with fixes to the image locations and the nav bar

FedG-code committed:
Completed all tasks

Titles are now accurate
Geo tags are now filled
The article list is now correctly passed through the template, eventually there will have to be code that makes the list file by itself with all the titles generated from the articles.

Final formatting problem is I can't seem to get the div around my banner to take the size of the banner, so that I can add a bottom margin and not cut off the top of the text in each page
